addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -25,3 +38,400 @@ def _compute_proxy_type(self):
+            max_num = invoice_data.get('max_num', 0)
+            retrigger = retrigger or current_num == max_num > 0
+
+            # Import the move from the attachment.
+            # `_create_document_from_attachment` will create an empty move
+            # then try and fill it with the content imported from the attachment.
+            # Should the import fail, thanks to try..except and savepoint,
+            # we will anyway end up with an empty `in_invoice` with the attachment posted on it.
+            # The move will be committed anyway
+            Journal._create_document_from_attachment(attachment.ids)
+            self.env.cr.commit()
+            proxy_acks.append(id_transaction)
+
+        return {"retrigger": retrigger, "proxy_acks": proxy_acks}
+
+    def _l10n_it_edi_get_moves_needing_update(self):
+        self.ensure_one()
+        moves_to_update = self.env['account.move'].search([
+            ('company_id', '=', self.company_id.id),
+            ('l10n_it_edi_transaction', '!=', False),
+            ('l10n_it_edi_state', 'in', WAITING_STATES),
+        ])
+        return {m.l10n_it_edi_transaction: m for m in moves_to_update}
+
+    def _l10n_it_edi_demo_mode_update(self, move=None):
+        self.ensure_one()
+        id_transaction_to_move_map = {move.l10n_it_edi_transaction: move} if move else self._l10n_it_edi_get_moves_needing_update()
+        for id_transaction, move_to_update in id_transaction_to_move_map.items():
+            filename = move_to_update.l10n_it_edi_attachment_id and move_to_update.l10n_it_edi_attachment_id.name or '???'
+            self._l10n_it_edi_process_update(
+                move_to_update,
+                id_transaction,
+                transformed_update_data={
+                    'l10n_it_edi_state': 'forwarded',
+                    'l10n_it_edi_transaction': f'demo_{uuid.uuid4()}',
+                    'send_ack_to_edi_proxy': False,
+                    'date': fields.Date.today(),
+                    'filename': filename},
+                message=_("The e-invoice file %s has been sent in Demo EDI mode.", filename))
+
+    def _l10n_it_edi_update(self, move=None):
+        ''' Check if the sent invoices have been processed by the SdI.
+        '''
+        self.ensure_one()
+        self = self.with_company(self.company_id)
+
+        if self.edi_mode == 'demo':
+            return self._l10n_it_edi_demo_mode_update(move)
+
+        server_url = self._get_server_url()
+        id_transaction_to_move_map = {move.l10n_it_edi_transaction: move} if move else self._l10n_it_edi_get_moves_needing_update()
+
+        try:
+            updates_data = self._make_request(
+                f'{server_url}/api/l10n_it_edi/1/in/TrasmissioneFatture',
+                params={'ids_transaction': list(id_transaction_to_move_map)})
+        except AccountEdiProxyError as e:
+            _logger.exception("Cannot update e-invoices: %s", e)
```

Comment:
Hum, nice questions that come from the use of the `Check now` button.
This used to be cron only, so I didn't think it through probably 👍🏻 
The button should probably have a UserError instead.
If the cron fails, before this PR, we had no feedback so we can also have another PR later to add some.

Reply:
I've added the IAP branch.

Reply:
```suggestion
            _logger.exception("Cannot update e-invoices: %s", e)
            return
```
Would this be the correct thing to do? Since updates_data isn't set, the method can't really continue.

Reply:
I have addressed these issues. Now the AccountEdiProxyErrors in contacting the IAP are raised as UserErrors, which are:
1) shown in the interface if called by the user
2) logged if raised in the CRON by the cron framework
If there's a traceback, we let it be.